components/applications-service/cmd/applications-service/commands/serve.go

```diff
@@ -56,7 +56,10 @@ var serveCmd = &cobra.Command{
 			)
 
 			// Trigger NATS Subscription
-			natsClient.ConnectAndSubscribe()
+			err := natsClient.ConnectAndSubscribe()
+			if err != nil {
+				log.WithError(err).Fatal("could not connect to nats-streaming")
```

Comment:
This will make applications service crash if it can't connect to NATS, which means we can't silently fail to start ingesting messages any more.

Reply:
@stevendanna this is the place where I think we are now bouncing constantly trying to connect to the NATS server, the server won't come up unless the user enables the feature flag so we will be in an infinite loop.

I can add a feature flag to avoid trying to connect unless the flag is enabled.